File: Misc/D_Product_of_Binary_Decimals.py
binary_decimals = []


def generate(n, ls):
    if n == 0:
        binary_decimals.append(int("".join(map(str, ls))))
        return
    generate(n - 1, ls + [0])
    generate(n - 1, ls + [1])


generate(4, [])
binary_decimals.remove(1)
binary_decimals.remove(0)

t = int(input())
for _ in range(t):
    n = int(input())

    # find a combination of numbers in binary_decimals whose product is n
    def find_combination(n):
        if n == 1:
            return True
        res = False
        for option in binary_decimals:
            if n % option == 0:
                res |= find_combination(n / option)
        return res

    if find_combination(n):
        print("YES")
    else:
        print("NO")

File: Misc/D_Product_of_Binary_Decimals.cpp
#include <iostream>
#include <vector>
#include <unordered_set>
#include <algorithm>

using namespace std;

unordered_set<int> binary_decimals;

void generate(int n, vector<int> ls)
{
    if (n == 0)
    {
        int num = 0;
        for (int i = 0; i < ls.size(); i++)
        {
            num = num * 10 + ls[i];
        }
        binary_decimals.insert(num);
        return;
    }
    ls.push_back(0);
    generate(n - 1, ls);
    ls.back() = 1;
    generate(n - 1, ls);
}

bool find_combination(int n)
{
    if (n == 1)
    {
        return true;
    }
    for (auto option : binary_decimals)
    {
        if (n % option == 0 && find_combination(n / option))
        {
            return true;
        }
    }
    return false;
}

int main()
{
    generate(4, {});
    binary_decimals.erase(std::remove(binary_decimals.begin(), binary_decimals.end(), 0), binary_decimals.end());
    int t;
    cin >> t;
    for (int i = 0; i < t; i++)
    {
        int n;
        cin >> n;
        if (find_combination(n))
        {
            cout << "YES" << endl;
        }
        else
        {
            cout << "NO" << endl;
        }
    }
    return 0;
}